app/src/main/java/at_challenge/deaguiar/daniel/atchalleng/controller/RouteDetailActivity.java

```diff
@@ -0,0 +1,203 @@
+import at_challenge.deaguiar.daniel.atchalleng.service.DownloadIntentService;
+import at_challenge.deaguiar.daniel.atchalleng.service.DownloadResultReceiver;
+import at_challenge.deaguiar.daniel.atchalleng.model.BusStop;
+import at_challenge.deaguiar.daniel.atchalleng.model.Departure;
+import at_challenge.deaguiar.daniel.atchalleng.model.DepartureList;
+import at_challenge.deaguiar.daniel.atchalleng.R;
+
+/**
+ * RouteDetailsActivity
+ * Responsible to fetch route data from server and show it to the user
+ *
+ * @author Daniel Besen de Aguiar
+ */
+public class RouteDetailActivity extends Activity implements DownloadResultReceiver.Receiver {
+
+    public static final String EXTRA_ROUTE_ID   = "id";
+    public static final String EXTRA_ROUTE_NAME = "longName";
+    private final static String END_POINT_STOPS = "https://api.appglu.com/v1/queries/findStopsByRouteId/run";
+    private final static String END_POINT_DEPARTURES = "https://api.appglu.com/v1/queries/findDeparturesByRouteId/run";
+
+    private int mRouteId;
+    private String mRouteName;
+    private BusStopList mBusStopList;
+    private DepartureList mDepartureList;
+
+    private TextView mRouteNameTextView;
+    private ListView mStopListView;
+    private ListView mWeekdayListView;
+    private ListView mStaturdayListView;
+    private ListView mSundayListView;
+
+    private DownloadResultReceiver mResultReceiver;
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_route_detail);
+
+        mRouteId   = getIntent().getIntExtra(EXTRA_ROUTE_ID, 0);
+        mRouteName = getIntent().getStringExtra(EXTRA_ROUTE_NAME);
+
+        ActionBar ab = getActionBar();
+        ab.setDisplayHomeAsUpEnabled(true);
+        ab.setTitle(R.string.route_and_departure);
+
+        mRouteNameTextView = (TextView)findViewById(R.id.route_detail_name);
+        mStopListView      = (ListView)findViewById(R.id.route_detail_stop_list);
+        mWeekdayListView   = (ListView)findViewById(R.id.route_detail_departure_list_weekday);
+        mStaturdayListView = (ListView)findViewById(R.id.route_detail_departure_list_saturday);
+        mSundayListView    = (ListView)findViewById(R.id.route_detail_departure_list_sunday);
+
+        // Prepare for fetching data from Server
+        mResultReceiver = new DownloadResultReceiver(new Handler());
+        mResultReceiver.setReceiver(this);
+
+        // Setting intent for DownloadResultReceiver
+        Intent intent = new Intent(Intent.ACTION_SYNC, null, this, DownloadIntentService.class);
+        intent.putExtra(DownloadIntentService.INTENT_ID, mRouteId);
+        intent.putExtra(DownloadIntentService.URL_BUS_STOP, END_POINT_STOPS);
+        intent.putExtra(DownloadIntentService.URL_BUS_DEPARTURE, END_POINT_DEPARTURES);
+        intent.putExtra("receiver", mResultReceiver);
+
+        startService(intent);
```

Comment:
- I would make classes BusStopList and DepartureList implements Serilizable
interface;
- save both lists in a Bundle using the method "onSaveInstanceState(Bundle
...)".
- Would check if Bundle is null in the onCreate(Bundle method). If is not
null, read from Bundle, else, start IntentService
- Create a method to update the ListAdapters.

2015-06-17 13:09 GMT-03:00 André Luiz Cardoso <notifications@github.com>:

> In
> app/src/main/java/at_challenge/deaguiar/daniel/atchalleng/controller/RouteDetailActivity.java
> <https://github.com/daniboy000/at_challenge/pull/1#discussion_r32643728>:
>
> > +        mWeekdayListView   = (ListView)findViewById(R.id.route_detail_departure_list_weekday);
> > +        mStaturdayListView = (ListView)findViewById(R.id.route_detail_departure_list_saturday);
> > +        mSundayListView    = (ListView)findViewById(R.id.route_detail_departure_list_sunday);
> > +
> > +        // Prepare for fetching data from Server
> > +        mResultReceiver = new DownloadResultReceiver(new Handler());
> > +        mResultReceiver.setReceiver(this);
> > +
> > +        // Setting intent for DownloadResultReceiver
> > +        Intent intent = new Intent(Intent.ACTION_SYNC, null, this, DownloadIntentService.class);
> > +        intent.putExtra(DownloadIntentService.INTENT_ID, mRouteId);
> > +        intent.putExtra(DownloadIntentService.URL_BUS_STOP, END_POINT_STOPS);
> > +        intent.putExtra(DownloadIntentService.URL_BUS_DEPARTURE, END_POINT_DEPARTURES);
> > +        intent.putExtra("receiver", mResultReceiver);
> > +
> > +        startService(intent);
>
> You are fetching the data from the server every time the activity is
> created. This causes the data to fetched again if the screen orientation
> changes and a new activity is created. How would you improve that?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/daniboy000/at_challenge/pull/1/files#r32643728>.
>



-- 
Daniel Besen de Aguiar

"Those who wish to succeed must ask the right preliminary questions"


      Aristotle